### PR TITLE
Add an extras_require section to simplify doc and test dependency installation

### PR DESCRIPTION
**Pull request by pelson**
I needed to build the docs, and it wasn't clear that I needed ``sphinxcontrib.bibtex``.
With this change you can ``pip install psyclone[doc]`` and get the dependencies needed (minus ``latexmk``, which is never going to pip install).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -100,6 +100,10 @@
         package_dir={"": "src"},
         install_requires=['pyparsing', 'fparser==0.0.8', 'configparser',
                           'six'],
+        extras_require={
```
Comment (arporter) [inline setup.py]: Ooh, I live and learn!

**Comment by pelson** [comment]
Thanks @arporter. The documentation is pretty comprehensive in the getting going section - perhaps overly so as it risks becoming dated quickly (best practice wrt. to python packaging is a bit of a moving target). I avoided addressing this here, but perhaps something to consider for the future.

I've now updated the PR based on your review.

File: doc/getting_going.rst [hunk 2]
```diff
@@ -306,6 +293,13 @@ the config file.
 Test
 ----
 
+You can install the necessary dependencies to run the PSYclone tests with::
+
+    > pip install psyclone[test]
+
+The test dependencies are canonically documented in PSYclone's setup.py
```
Comment (arporter) [inline doc/getting_going.rst]: s/PSYclone/PSyclone/

File: doc/getting_going.rst [hunk 3]
```diff
@@ -42,18 +40,7 @@ Dependencies
 ------------
 
 PSyclone is written in Python so needs Python to be installed on the
-target machine. PSyclone has been tested under Python 2.6.5, 2.7.3 and 3.6.
-
-.. warning:: Release 1.6.0 of PSyclone requires version 0.0.7 of
```
Comment (arporter) [inline doc/getting_going.rst]: Please put back this warning.

**Comment by pelson** [comment]
I've reverted the warning about the previous version of PSyclone, so hopefully this is now good to go. Please could you take another look @arporter?

**Comment by arporter** [comment]
Thanks for putting back the warning. I've taken the liberty of fixing some formatting errors.

**Comment by pelson** [comment]
Thanks @arporter. 👍

**Pull request merged**